src/Microsoft.AspNetCore.Mvc.RazorPages/ApplicationModels/PageHandlerModel.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Mvc.ApplicationModels
+{
+    /// <summary>
+    /// Represents a handler in a <see cref="PageApplicationModel"/>.
+    /// </summary>
+    [DebuggerDisplay("PageHandlerModel: Name={" + nameof(PageHandlerModel.Name) + "}")]
+    public class PageHandlerModel : ICommonModel
+    {
+        /// <summary>
+        /// Creates a new <see cref="PageHandlerModel"/>.
+        /// </summary>
+        /// <param name="handlerMethod">The <see cref="System.Reflection.MethodInfo"/> for the handler.</param>
+        /// <param name="attributes">Any attributes annotated on the handler method.</param>
+        public PageHandlerModel(
+            MethodInfo handlerMethod,
+            IReadOnlyList<object> attributes)
+        {
+            MethodInfo = handlerMethod ?? throw new ArgumentNullException(nameof(handlerMethod));
+            Attributes = attributes ?? throw new ArgumentNullException(nameof(attributes));
+
+            Parameters = new List<PageParameterModel>();
+            Properties = new Dictionary<object, object>();
+        }
+
+        /// <summary>
+        /// Creats a new instance of <see cref="PageHandlerModel"/> from a given <see cref="PageHandlerModel"/>.
+        /// </summary>
+        /// <param name="other">The <see cref="PageHandlerModel"/> which needs to be copied.</param>
+        public PageHandlerModel(PageHandlerModel other)
+        {
+            if (other == null)
+            {
+                throw new ArgumentNullException(nameof(other));
+            }
```

Comment:
Well, that's the annoying bit - it only works for assignments, not regular null checks. Super inconsistent. Sort of one of the reasons I wasn't a big fan of it.